src/DotNetty.Transport/Channels/Groups/DefaultChannelGroupCompletionSource.cs

```diff
@@ -34,7 +34,7 @@ public DefaultChannelGroupCompletionSource(IChannelGroup group, Dictionary<IChan
                 this.futures.Add(pair.Key, pair.Value);
                 pair.Value.ContinueWith(x =>
                 {
-                    bool success = x.IsCompleted && !x.IsFaulted && !x.IsCanceled;
+                    bool success = x.Status == TaskStatus.RanToCompletion;
```

Comment:
correct, only difference is that we're doing a single volatile read with the updated version.